### PR TITLE
CartController testing

### PR DESCRIPTION
**Pull request by joshsadam**
## Description of changes
Re-implement testing for the `CartController`. There is a bunch of code removal that was dead code from this cart update found while implementing these tests.

## Related issue
#192 

## Checklist
Things for the developer to confirm they've done before the PR should be accepted:

~~* [ ] CHANGELOG.md (and UPGRADING.md if necessary) updated with information for new change.~~
~~* [ ] Tests added (or description of how to test) for any new features.~~
~~* [ ] User documentation updated for UI or technical changes.~~

### CODE REVIEW

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java [hunk 1]
```diff
@@ -1,114 +0,0 @@
...
-@ContextConfiguration(loader = AnnotationConfigWebContextLoader.class, classes = { IridaApiJdbcDataSourceConfig.class,
-		IridaApiPropertyPlaceholderConfig.class, IridaApiServicesConfig.class,
-		IridaUIWebConfig.class, IridaWebTestScopeConfig.class })
-@TestExecutionListeners({ DependencyInjectionTestExecutionListener.class, DbUnitTestExecutionListener.class,
-		WithSecurityContextTestExcecutionListener.class })
-@ActiveProfiles("it")
-@WebAppConfiguration
-@DatabaseSetup("/ca/corefacility/bioinformatics/irida/ria/web/analysis/CartControllerIT.xml")
-@DatabaseTearDown("classpath:/ca/corefacility/bioinformatics/irida/test/integration/TableReset.xml")
-public class CartControllerIT {
```
Comment (apetkau) [inline src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java]: Are these tests not relevant anymore? Will they be included in another merge request?

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/cart/CartController.java [hunk 2]
```diff
@@ -15,22 +10,14 @@
...
 import ca.corefacility.bioinformatics.irida.model.project.Project;
 import ca.corefacility.bioinformatics.irida.model.sample.Sample;
-import ca.corefacility.bioinformatics.irida.model.sample.SampleSequencingObjectJoin;
-import ca.corefacility.bioinformatics.irida.model.sequenceFile.SequenceFile;
 import ca.corefacility.bioinformatics.irida.ria.web.cart.components.Cart;
 import ca.corefacility.bioinformatics.irida.ria.web.cart.dto.*;
 import ca.corefacility.bioinformatics.irida.service.ProjectService;
 import ca.corefacility.bioinformatics.irida.service.SequencingObjectService;
 import ca.corefacility.bioinformatics.irida.service.sample.SampleService;
 import ca.corefacility.bioinformatics.irida.service.user.UserService;
```
Comment (apetkau) [inline src/main/java/ca/corefacility/bioinformatics/irida/ria/web/cart/CartController.java]: It looks like both `UserService` and `SequencingObjectService` are unused in this class now. Could you remove those?

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java (hunk 1 above)
Comment (joshsadam) [reply src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java]: Yeah, I already have a merge request about 2 away that specifically tests the new cart functionality.

File: src/main/java/ca/corefacility/bioinformatics/irida/ria/web/cart/CartController.java (hunk 2 above)
Comment (joshsadam) [reply src/main/java/ca/corefacility/bioinformatics/irida/ria/web/cart/CartController.java]: Thanks, updated in 4df5132465c06c43fa35c81b19c454c0880efb7e

File: src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java (hunk 1 above)
Comment (apetkau) [reply src/test/java/ca/corefacility/bioinformatics/irida/ria/integration/analysis/CartControllerIT.java]: Great.

**Comment by apetkau** [comment]
Thanks @joshsadam. I'm merging this in.

**Pull request merged**